src/primitives/block.cpp

```diff
@@ -109,6 +112,47 @@ uint256 CBlock::CheckMerkleBranch(uint256 hash, const std::vector<uint256>& vMer
     return hash;
 }
 
+static uint64_t next_pow2(uint64_t x)
+{
+    x -= 1;
+    x |= (x >> 1);
+    x |= (x >> 2);
+    x |= (x >> 4);
+    x |= (x >> 8);
+    x |= (x >> 16);
+    x |= (x >> 32);
+    return x + 1;
+}
+
+uint256 CBlock::BuildAuthDataMerkleTree() const
+{
+    std::vector<uint256> tree;
+    auto perfectSize = next_pow2(vtx.size());
+    tree.reserve(perfectSize * 2); // Safe upper bound for the number of total nodes.
+
+    // Add the leaves to the tree. v1-v4 transactions will append empty leaves.
+    for (auto &tx : vtx) {
+        tree.push_back(tx.GetAuthDigest());
+    }
+    // Append empty leaves until we get a perfect tree.
+    tree.insert(tree.end(), perfectSize - vtx.size(), uint256());
```

Comment:
https://github.com/zcash/zips/issues/519#issuecomment-860709521

Reply:
There is an ambiguity between the zero hash used as a placeholder for the auth digest of a pre-v5 transaction, and the zero hashes used to fill the tree out to a power of two. This ambiguity would only affect a process that was just trying to check the auth digest tree, but it is worth fixing now. Kris and I decided that this should instead fill the tree with 0xFF.. placeholders.